### PR TITLE
IOS Prefix list resource module

### LINKED ISSUE #196: ios_prefix_lists resource module

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

IOS Prefix list resource module. Fixes #196 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ios_prefix_lists

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by justjais** [comment]
recheck

File: plugins/module_utils/network/ios/rm_templates/prefix_lists.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            "name": "prefix_list",
+            "getval": re.compile(
+                r"""
+                ^(?P<afi>ip|ipv6)*
+                \s*prefix-list*
+                \s*(?P<name>\S+)*
+                \s*(?P<description>description\s\S.*)*
+                \s*(?P<sequence>seq\s\S+)*
+                \s*(?P<action>deny|permit)*
+                \s*(?P<address>(?:[0-9]{1,3}\.){3}[0-9]{1,3}/\d+|(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/\d+)*
```
Comment (NilashishC) [inline plugins/module_utils/network/ios/rm_templates/prefix_lists.py:58]: This IPv4/IPv6 validation regex is redundant. We are reading this from a device which wouldn't have accepted a wrong IPv4/IPv6.

File: changelogs/fragments/192_add_prefix_lists_resource_module.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
```
Comment (NilashishC) [inline changelogs/fragments/192_add_prefix_lists_resource_module.yaml:2]: This went in by mistake?

File: plugins/modules/ios_prefix_lists.py [hunk 3]
```diff
@@ -0,0 +1,894 @@
...
+                description: Specify packets to be rejected or forwarded
+                type: str
+                choices: ['deny', 'permit']
+              sequence:
+                description: sequence number of an entry
+                type: int
+              description:
+                description:  Prefix-list specific description
+                type: str
+              address:
```
Comment (NilashishC) [inline plugins/modules/ios_prefix_lists.py:60]:
```suggestion
              prefix:
```

**Review by NilashishC — APPROVED** [review]

**Pull request merged**